modules/photo/src/ccm/utils.cpp

```diff
@@ -11,14 +11,23 @@
 namespace cv {
 namespace ccm {
 
-inline double gammaCorrection_(double element, double gamma)
+inline double gammaOp(double element, double gamma)
 {
-    return (element >= 0 ? pow(element, gamma) : -pow((-element), gamma));
+    return (element >= 0.0) ? pow(element, gamma) : -pow(-element, gamma);
 }
 
-Mat gammaCorrection(const Mat& src, double gamma, Mat dst)
+void gammaCorrection(InputArray _src, OutputArray _dst, double gamma)
 {
-    return elementWise(src, [gamma](double element) -> double { return gammaCorrection_(element, gamma); }, dst);
+    Mat src = _src.getMat();
```

Comment:
please add CV_Assert with input type checks at least. The function is public API now and should handle invalid input correctly.